### DISCUSSION TITLE
Importing using Blocks

### DISCUSSION THREAD

**Discussion by zoligyenge** [pybricks/support#1939] [General]
Dear Community.

I was wondering whether it is possible to make a call to an imported my_task which uses a device (e.g. a drive base) set up in the main program?

The only way I found was to pass all defined  devices to the imported task, but that is quite combersome for complex tasks.

![import](https://github.com/user-attachments/assets/9763e92a-2838-455e-b5c4-16616fbb89d8)
![Main](https://github.com/user-attachments/assets/5910831c-be67-4fb9-83be-d0c313d5768a)

**Comment by BertLindeman** [comment]
Not very stable idea but I think you could make a variable with a list of the devices you need to pass.
Then it is simply passing that list.

Maybe too simple, just a thought after diner.

Bert

**Comment by laurensvalk** [comment]
What you're doing does seem like the right approach.

Another approach with fewer blocks overal would be to use global variables.

So in your "library program" (your first picture) you could make a variable (red block) called drive base and leave it at 0. Then make a function called "set drive base" and pass the drive base object to it. This is much like you already do, but then you only have to make one such function. This function will set the drivebase variable to the drivebase you just passed in.

Then you can make any other number of drive functions without passing in the drivebase object again. They can use the variable instead.
